src/main/java/edu/harvard/iq/dataverse/export/ExportService.java

```diff
@@ -201,157 +252,168 @@ public void exportAllFormats(Dataset dataset) throws ExportException {
+                } else {
+                    cacheExport(dataset, dataProvider, formatName, e);
+                }
             }
-            // Finally, if we have been able to successfully export in all available 
-            // formats, we'll increment the "last exported" time stamp: 
+            // Finally, if we have been able to successfully export in all available
+            // formats, we'll increment the "last exported" time stamp:
             dataset.setLastExportTime(new Timestamp(new Date().getTime()));
-            
+
         } catch (ServiceConfigurationError serviceError) {
             throw new ExportException("Service configuration error during export. " + serviceError.getMessage());
         } catch (RuntimeException e) {
-            //e.printStackTrace();
-            throw new ExportException("Unknown runtime exception exporting metadata. " + (e.getMessage() == null ? "" : e.getMessage()));
+            // e.printStackTrace();
+            throw new ExportException(
+                    "Unknown runtime exception exporting metadata. " + (e.getMessage() == null ? "" : e.getMessage()));
         }
 
     }
 
     public void clearAllCachedFormats(Dataset dataset) throws IOException {
         try {
-            Iterator<Exporter> exporters = loader.iterator();
-            while (exporters.hasNext()) {
-                Exporter e = exporters.next();
-                String formatName = e.getProviderName();
 
+            for (Exporter e : exporterMap.values()) {
+                String formatName = e.getFormatName();
                 clearCachedExport(dataset, formatName);
             }
 
             dataset.setLastExportTime(null);
         } catch (IOException ex) {
-            //not fatal
+            // not fatal
         }
     }
 
-    // This method finds the exporter for the format requested, 
+    // This method finds the exporter for the format requested,
     // then produces the dataset metadata as a JsonObject, then calls
-    // the "cacheExport()" method that will save the produced output  
-    // in a file in the dataset directory. 
+    // the "cacheExport()" method that will save the produced output
+    // in a file in the dataset directory.
     public void exportFormat(Dataset dataset, String formatName) throws ExportException {
         try {
-            Iterator<Exporter> exporters = loader.iterator();
-            while (exporters.hasNext()) {
-                Exporter e = exporters.next();
-                if (e.getProviderName().equals(formatName)) {
-                    DatasetVersion releasedVersion = dataset.getReleasedVersion();
-                    if (releasedVersion == null) {
-                        throw new ExportException("No published version found during export. " + dataset.getGlobalId().toString());
+
+            Exporter e = exporterMap.get(formatName);
+            if (e != null) {
+                DatasetVersion releasedVersion = dataset.getReleasedVersion();
```

Comment:
This pins the export to `getReleasedVersion` would it make sense to just export the passed Dateset and move the responsible to select the latest/released version to the caller?

Reply:
Also it might look new code, this is the behaviour as it has been for a long time. Not meaning to say this is good, but changing this is beyond scope for this PR. (Needs new use case, issue, PR, funding, ...)

Reply:
FWIW: I think this is one of the reasons that the internal representations are used to create the exports and not vice versa (see [other comment](https://github.com/IQSS/dataverse/pull/9175#discussion_r1192435908)) - the underlying code can create representations for different versions. A new PR that provides version-specific inputs to the exporter would be a good step in that direction. (I can't find it now but someone did try to refactor the getJsonLD (schema.org output) method and got stuck here (the job turned out to be bigger because exports aren't version-specific).)